docs/core/pipes/type-cast.pipe.md

```diff
@@ -0,0 +1,36 @@
+---
+Title: Type cast pipe
+Added: v6.0.0
+Status: Active
+Last reviewed: 2023-05-23
+---
+
+# [Type Cast pipe](../../../lib/core/src/lib/pipes/type-cast.pipe.ts "Defined in user-initial.pipe.ts")
+
+Takes a reference to a constructor and then type casts the provided object to the type of that constructor. 
+
+## Basic Usage
+
+<!-- {% raw %} -->
+```TS
+    protected readonly MyCustomTypeConstructor = MyCustomTypeConstructor
+```
+```HTML
+<div>
+    <my-comp [inputProp]="someVar | cast : MyCustomTypeConstructor"></my-comp>
```

Comment:
this does not fix the issue of the component not using proper typing in the properties. Angular already provides "$any" cast for cases when one needs to suppress the warning:
```
[node]="$any(site)"
```

Reply:
The function call of $any() is the reason why we created this pipe. The benefit of this pipe is that it allows us to assert an object to a specifc type, without requiring the use of the $any function in the template. 

If we are fine with using $any(), then we don't need this new pipe at all.